### PR TITLE
fix(cluster): fix slot filtration to RestoreStreamer

### PR DESCRIPTION
**Pull request by BorysTheDev**

### CODE REVIEW

File: src/server/journal/streamer.cc [hunk 1]
```diff
@@ -123,12 +127,14 @@ void RestoreStreamer::WriteBucket(PrimeTable::bucket_iterator it) {
...
   {
     FiberAtomicGuard fg;  // Can't switch fibers because that could invalidate iterator
-
+    string key_buffer;    // we can reuse it
     while (!it.is_done()) {
       const auto& pv = it->second;
 
-      string key_buffer;
       string_view key = it->first.GetSlice(&key_buffer);
+      if (!ShouldWrite(key)) {
```
Comment (adiholden) [inline src/server/journal/streamer.cc:135]: If no slot from the bucket was written as the bucket did not contain any slot data we should not call NotifyWritten below

File: tests/dragonfly/cluster_test.py [hunk 2]
```diff
@@ -835,5 +835,128 @@ async def test_cluster_slot_migration(df_local_factory: DflyInstanceFactory):
...
+        "DFLYCLUSTER", "START-SLOT-MIGRATION", "127.0.0.1", str(nodes[0].admin_port), "3000", "9000"
+    )
+    assert "OK" == res
+
+    await asyncio.sleep(1)
+
+    status = await c_nodes_admin[1].execute_command(
+        "DFLYCLUSTER", "SLOT-MIGRATION-STATUS", "127.0.0.1", str(nodes[0].admin_port)
+    )
+    assert "STABLE_SYNC" == status
```
Comment (chakaz) [inline tests/dragonfly/cluster_test.py:932]: Can we issue `SLOT-MIGRATION-STATUS` in a loop, without sleeping before, and with a short sleep in between? And finish the loop when we reach stable sync
That will make the test both faster and more stable

File: tests/dragonfly/cluster_test.py [hunk 3]
```diff
@@ -835,5 +835,128 @@ async def test_cluster_slot_migration(df_local_factory: DflyInstanceFactory):
...
+        config.replace("LAST_SLOT_CUTOFF", "5259").replace("NEXT_SLOT_CUTOFF", "5260"),
+        c_nodes_admin,
+    )
+
+    await c_nodes_admin[0].close()
+    await c_nodes_admin[1].close()
+
+
+@dfly_args({"proactor_threads": 4, "cluster_mode": "yes"})
+async def test_cluster_data_migration(df_local_factory: DflyInstanceFactory):
```
Comment (chakaz) [inline tests/dragonfly/cluster_test.py:848]: Do we need 2 tests for data and slot migration? Why not do both in the same test? What's the benefit of having 2 tests?

File: tests/dragonfly/cluster_test.py (hunk 2 above)
Comment (BorysTheDev) [reply tests/dragonfly/cluster_test.py:932]: I think we can

File: tests/dragonfly/cluster_test.py (hunk 3 above)
Comment (BorysTheDev) [reply tests/dragonfly/cluster_test.py:848]: I don't want to have a really big test. in cluster_migration_test I check corner cases of API and in current test data.

**Review by adiholden — APPROVED** [review]
